backend/api/src/realms.rs

```diff
@@ -29,10 +32,31 @@ impl Realm {
         Id::realm(self.parent_key)
     }
 
+    fn path(&self) -> &str {
+        &self.path
+    }
+
     fn parent(&self, context: &Context) -> &Realm {
         &context.realm_tree.realms[&self.parent_key]
     }
 
+    fn parents(&self, context: &Context) -> Vec<&Realm> {
+        let mut parents = VecDeque::new();
+        let mut generation = self;
+
+        // Add all the parents up to the root (which might be ourselves already,
+        // in which case this never runs)
+        while generation.parent_key != 0 {
+            generation = context.realm_tree.realms.get(&generation.parent_key).unwrap();
+            parents.push_front(generation);
+        }
+        // Always add the root
+        generation = context.realm_tree.realms.get(&generation.parent_key).unwrap();
+        parents.push_front(generation);
+
+        parents.into()
```

Comment:
Instead of using a `VecDeque` and converting it into a `Vec` here, why not just use a `Vec` and accept the reversed order? Like, I wouldn't say one order is clearly better/correct. `parents: [parent, grandparent, root]` is equally useful IMO.

Reply:
Well, there is one case where we need this currently, and it requires a specific order: The breadcrumbs. So I thought I might as well provide the correct order. I have no hard opinion, though. If you want, we can also invert it on the client.

Reply:
Yes, I would say so. The frontend surely needs it in another order at some point, too. So let's just keep the backend simple then.

Reply:
Since we now know about two places that need the other order, I really think the endpoint should return that. However, the iterator thing we talked about still returns the order that is easier for it to generate, and I even use that fact for an optimization.

Dunno, I have no hard opinion beyond "this PR is how I would do it, after discussing a bit further with you and also thinking about stuff more and actually implementing things". Some things will always be awkward.

Reply:
Yeah no, totally ok for now